modular_skyrat/modules/modular_implants/code/soulcatcher/handheld_soulcatcher.dm

```diff
@@ -47,68 +56,112 @@
+		return FALSE
 
-		if(!target_mob.GetComponent(/datum/component/previous_body))
+	var/turf/source_turf = get_turf(user)
+	var/admin_log = "[key_name(user)] used [src] to put [key_name(target_mob)]'s mind into a soulcatcher at [AREACOORD(source_turf)]."
+	if (ghost)
+		if(!target_room.add_soul_from_ghost(ghost))
 			return FALSE
+	else
+		admin_log += " [target_mob] was still alive."
+		target_room.add_soul(target_mob.mind, TRUE)
 
-		var/turf/source_turf = get_turf(user)
-		log_admin("[key_name(user)] used [src] to put [key_name(target_mob)]'s mind into a soulcatcher at [AREACOORD(source_turf)]")
-		linked_soulcatcher.scan_body(target_mob, user)
-		return TRUE
+		playsound(src, 'modular_skyrat/modules/modular_implants/sounds/default_good.ogg', 50, FALSE, ignore_walls = FALSE)
+		visible_message(span_notice("[src] beeps: [target_mob]'s mind transfer is now complete."))
 
-	var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Choose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms, timeout = 30 SECONDS)
-	if(!target_room)
+	log_admin(admin_log)
+
+	if(!target_mob.GetComponent(/datum/component/previous_body))
 		return FALSE
+	linked_soulcatcher.scan_body(target_mob, user)
 
-	SEND_SOUND(target_mob, 'sound/misc/notice2.ogg')
-	window_flash(target_mob.client)
+	return TRUE
 
-	if((tgui_alert(target_mob, "Do you wish to enter [target_room]? This will remove you from your body until you leave.", name, list("Yes", "No"), 30 SECONDS, FALSE) != "Yes") || (tgui_alert(target_mob, "Are you sure about this?", name, list("Yes", "No"), 30 SECONDS, FALSE) != "Yes"))
-		COOLDOWN_START(src, rsd_scan_cooldown, RSD_ATTEMPT_COOLDOWN)
-		to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+/**
+ * Invites target_mob into target_room by giving them a tgui_alert.
+ * target_mob cannot be invited more than once while a alert from a given soulcatcher is open, enforced by
+ * src.confirming_entry += real_target and a check surrounding that.
+ *
+ * Args:
+ * datum/soulcatcher/target_room: The room we are inviting target_mob to.
+ * mob/living/user (Nullable): The individual who invited the mob.
+ * mob/living/target_mob: The mob that was clicked on/invited into the room.
+ *
+ * Returns:
+ * False if the invitation wasn't delivered, the result of the tgui_alert otherwise.
+ */
+/obj/item/handheld_soulcatcher/proc/invite_soul(datum/soulcatcher_room/target_room, mob/living/user, mob/living/target_mob)
+	if (target_mob in confirming_entry)
+		if (user)
+			to_chat(user, span_warning("You've already invited this person to a room, wait for them to respond!"))
 		return FALSE
 
-	if(!target_mob.mind)
-		return FALSE
+	var/message = "Do you want to transfer your soul into [target_room]?"
+	if (user)
+		message = "[user] wants to transfer you to [target_room] inside of a soulcatcher, do you accept?"
+	var/target_dead = (target_mob.stat == DEAD)
```

Comment:
You don't have to be allergic to spacing out your code, clean code is code that breathes, and that's spaced out properly and logically, so it's easier to read through.
```suggestion

	var/target_dead = (target_mob.stat == DEAD)
```

Reply:
hey i did space it out, i just usually dont go with multi-line spacers
ill put it in though